module/input/SensorFilter/src/SensorFilter.cpp

```diff
@@ -772,6 +784,32 @@ namespace module::input {
                             Hwt.translation() = o.rTWw;
                             sensors->Htw      = Hwt.inverse().matrix();
 
+                            // If there is ground truth data, determine the error in the odometry calculation
+                            // and emit graphs of those errors
+                            if (input.odometry_ground_truth.exists) {
+                                Eigen::Affine3d true_Htw(input.odometry_ground_truth.Htw);
+
+                                // Determine translational distance error
+                                Eigen::Vector3d est_rWTt   = Hwt.inverse().translation();
+                                Eigen::Vector3d true_rWTt  = true_Htw.translation();
+                                Eigen::Vector3d error_rWTt = (true_rWTt - est_rWTt).cwiseAbs();
+
+                                // Determine yaw, pitch and roll error
+                                Eigen::Vector3d true_Rtw  = MatrixToEulerIntrinsic(true_Htw.rotation());
+                                Eigen::Vector3d est_Rtw   = MatrixToEulerIntrinsic(Hwt.inverse().rotation());
+                                Eigen::Vector3d error_Rtw = (true_Rtw - est_Rtw).cwiseAbs();
```

Comment:
Euler angles aren't great for error since they are non linear and influence eachother. Better if you can to get a single value error here

Reply:
I feel like this PR is going around in circles

Reply:
I've added in a single value as well as the decomposed since people wanted both.